ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh

```diff
@@ -6,101 +6,113 @@ set -o pipefail
-oc image mirror -f /tmp/mirror --registry-config ${DEVSCRIPTS_WORKING_DIR}/pull_secret.json
+oc image mirror -f /tmp/mirror --registry-config ${DS_WORKING_DIR}/pull_secret.json
 EOF
+        TEST_ARGS="--from-repository ${DEVSCRIPTS_TEST_IMAGE_REPO}"
+}
 
-  TEST_ARGS="--from-repository ${DEVSCRIPTS_TEST_IMAGE_REPO}"
-
-  echo "### Enriching test-list cases"
-  cat "${SHARED_DIR}/test-list-ext" >> "${SHARED_DIR}/test-list"
-fi
+function use_minimal_test_list() {
+        echo "### Skipping test images mirroring, fall back to minimal tests list"
+        
+        TEST_ARGS="--file /tmp/tests"
+        TEST_SKIPS=""
+        echo "${TEST_MINIMAL_LIST}" > /tmp/tests
+}
 
-# Test upgrade for workflows that requested it
-if [[ "$RUN_UPGRADE_TEST" == true ]]; then
-    echo "### Running Upgrade tests"
-    timeout \
-    --kill-after 10m \
-    120m \
-        ssh \
-            "${SSHOPTS[@]}" \
-            "root@${IP}" \
-            openshift-tests \
-            run-upgrade \
-            ${TEST_ARGS} \
-            --to-image "$OPENSHIFT_UPGRADE_RELEASE_IMAGE" \
-            -o /tmp/artifacts/e2e-upgrade.log \
-            --junit-dir /tmp/artifacts/junit-upgrade \
-            platform
-else
-    if [[ -s "${SHARED_DIR}/test-list" ]]; then
-        echo "### Copying test-list file"
-        scp \
-            "${SSHOPTS[@]}" \
-            "${SHARED_DIR}/test-list" \
-            "root@${IP}:/tmp/test-list"
-        echo "### Running tests"
-        timeout \
-        --kill-after 10m \
-        120m \
-        ssh \
-            "${SSHOPTS[@]}" \
-            "root@${IP}" \
-            openshift-tests \
-            run \
-            "openshift/conformance" \
-            --dry-run \
-            \| grep -Ff /tmp/test-list \|openshift-tests run ${TEST_ARGS} -o /tmp/artifacts/e2e.log --junit-dir /tmp/artifacts/junit -f -
+case "${CLUSTER_TYPE}" in
+packet)
+    # shellcheck source=/dev/null
+    source "${SHARED_DIR}/packet-conf.sh"
+    # shellcheck source=/dev/null
+    source "${SHARED_DIR}/ds-vars.conf"
+    copy_test_binaries
+    export TEST_PROVIDER=\"\"
+
+    echo "### Checking release version"
+    # Mirroring test images is supported only for versions greater than or equal to 4.7
+    if printf '%s\n%s' "4.7" "${DS_OPENSHIFT_VERSION}" | sort -C -V; then
+        mirror_test_images       
     else
-        echo "### Running tests"
-        ssh \
-            "${SSHOPTS[@]}" \
-            "root@${IP}" \
-            openshift-tests \
-            run \
-            "openshift/conformance/parallel" \
-            --dry-run \
-            \| grep 'Feature:ProjectAPI' \| openshift-tests run ${TEST_ARGS} -o /tmp/artifacts/e2e.log --junit-dir /tmp/artifacts/junit -f -
+        use_minimal_test_list
     fi
-fi
+    ;;
+*) echo >&2 "Unsupported cluster type '${CLUSTER_TYPE}'"; exit 1;;
+esac
+
+function upgrade() {
+    set -x
+    timeout --kill-after 10m 120m \
```

Comment:
Is this timeout usage valid for the 'standard' workflows. Because I saw a couple of different places have different timeout values being set in this PR.

Reply:
No, it's a workaround added to gather some results when a test get stuck, hopefully it could be removed once the test failures will get fixed (and it should be properly sized to give enough time to the tests to complete in the standard case).